tools/main.cpp

```diff
@@ -0,0 +1,342 @@
+        }
+        while (fgets(l_buffer.data(), l_buffer.size(), l_cmdPipe.get()) !=
+               nullptr)
+        {
+            l_cmdOutput.emplace_back(l_buffer.data());
+        }
+
+        if (l_cmdOutput.size() > 0)
+        {
+            std::transform(l_cmdOutput[0].begin(), l_cmdOutput[0].end(),
+                           l_cmdOutput[0].begin(), [](unsigned char l_char) {
+                return std::tolower(l_char);
+            });
+
+            // Remove the new line character from the string.
+            l_cmdOutput[0].erase(l_cmdOutput[0].length() - 1);
+
+            return l_cmdOutput[0] == "fieldmode=true" ? true : false;
+        }
+    }
+    catch (const std::exception& l_ex)
+    {}
+
+    return false;
+}
+
+/**
+ * @brief An API to create a PEL.
+ *
+ * This API makes synchronous call to phosphor-logging Create method.
+ *
+ * @param[in] errorType - Event message name.
+ * @param[in] severity - Severity of the event.
+ * @param[in] fileName - File name.
+ * @param[in] description - Error description.
+ */
+void createPel(const std::string& errorType, const std::string& severity,
+               const std::string& fileName,
+               const std::string& description) noexcept
+{
+    try
+    {
+        if (errorType.empty() || severity.empty() || fileName.empty() ||
+            description.empty())
+        {
+            throw std::runtime_error(
+                "Input(s) passed to create PEL is found empty");
+        }
+
+        std::map<std::string, std::string> additionalData{
+            {"fileName", fileName}, {"DESCRIPTION", description}};
+
+        auto bus = sdbusplus::bus::new_default();
+        auto method = bus.new_method_call(
+            "xyz.openbmc_project.Logging", "/xyz/openbmc_project/logging",
+            "xyz.openbmc_project.Logging.Create", "Create");
+        method.append(errorType, severity, additionalData);
+        bus.call(method);
+    }
+    catch (const sdbusplus::exception::SdBusError& ex)
+    {
+        std::cerr << "PEL creation failed with an error: " +
+                         std::string(ex.what())
+                  << std::endl;
+    }
+}
+
+/**
+ * @brief API to log a PEL if just not help command.
+ *
+ * The API logs a PEL when led-tool being used in the system.
+ * PEL will be skipped in case help command used.
+ *
+ * @param[in] argc - Number of arguments recieved.
+ * @param[in] argv - Arrguments passed.
+ */
+void logPelIfNotHelp(int argc, char** argv) noexcept
+{
+    bool logPel = true;
+    std::string cmd{"user entered command: "};
+
+    for (auto count = 0; count < argc; count++)
+    {
+        cmd += argv[count];
+
+        if (argv[count] == std::string("--help") ||
+            argv[count] == std::string("-h") || argc == 1)
+        {
+            logPel = false;
+        }
+
+        cmd += (count < argc - 1) ? " " : "";
+    }
+
+    if (logPel)
+    {
+        createPel("xyz.openbmc_project.Common.Error.InternalFailure",
```

Comment:
We would need a unique PEL, not InternalFailure.  InternalFailure has a side affect of triggering a BMC dump.